uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -1089,32 +1092,32 @@ public InstanceTopicPartitionHolder createNewRoute(String pipeline, int routeId)
     }
 
     _availableControllerList.remove(instanceName);
-    _pipelineToInstanceMap.put(pipeline, new PriorityQueue<>(1,
+    _pipelineToInstanceMap.putIfAbsent(pipeline, new PriorityQueue<>(1,
         InstanceTopicPartitionHolder.totalWorkloadComparator(_pipelineWorkloadMap)));
     _pipelineToInstanceMap.get(pipeline).add(instance);
     _assignedControllerCount.inc();
     _workerHelixManager.addTopicToMirrorMaker(instance, pipeline, routeId);
 
     // register metrics
-    String routeString = srcDst[1] + "-" + srcDst[2] + "-" + routeId;
     maybeRegisterMetrics(routeString);
 
     spectator.disconnect();
     return instance;
   }
 
   public InstanceTopicPartitionHolder maybeCreateNewRoute(
-      PriorityQueue<InstanceTopicPartitionHolder> instanceList,
-      String topicName,
-      int numPartitions,
-      String pipeline) throws Exception {
+          PriorityQueue<InstanceTopicPartitionHolder> instanceList,
+          String topicName,
+          int numPartitions,
+          String pipeline) throws Exception {
 
     LOGGER.info("maybeCreateNewRoute, topicName: {}, numPartitions: {}, pipeline: {}", topicName, numPartitions,
-        pipeline);
+            pipeline);
 
     Set<Integer> routeIdSet = new HashSet<>();
     for (InstanceTopicPartitionHolder instance : instanceList) {
       if (instance.getTotalNumPartitions() + numPartitions < _initMaxNumPartitionsPerRoute) {
+        _workerHelixManager.expandPipelineToMirrorMaker(pipeline, instance.getRoute().getPartition());
```

Comment:
We don't need to adjust worker count when adding topics. Rebalance will take care of it.